addons/account/static/src/js/open_move_widget.js

```diff
@@ -1,23 +1,22 @@
 /** @odoo-module **/
-import fieldRegistry from 'web.field_registry';
-import { FieldChar } from 'web.basic_fields';
 
-const OpenMoveWidget = FieldChar.extend({
-    template: 'account.OpenMoveTemplate',
-    events: Object.assign({}, FieldChar.prototype.events, {
-        'click': '_onOpenMove',
-    }),
-    _onOpenMove: function(ev) {
-        ev.preventDefault();
-        ev.stopPropagation();
-        var self = this;
-        this._rpc({
-            model: 'account.move.line',
-            method: 'open_move',
-            args: [this.res_id],
-        }).then(function (actionData){
-            return self.do_action(actionData);
-        });
-    },
-});
-fieldRegistry.add('open_move_widget', OpenMoveWidget);
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+const { Component } = owl;
+
+class OpenMoveWidget extends Component {
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+        this.action = useService("action");
+    }
+
+    async openMove(ev) {
+        const action = await this.orm.call("account.move.line", "open_move", [this.props.record.resId], {});
+        this.action.doAction(action);
+    }
+}
+
+OpenMoveWidget.template = "account.OpenMoveWidget";
+registry.category("fields").add("open_move_widget", OpenMoveWidget);
```

Comment:
Wouldn't this ideally be a generic field widget 
* on m2o fields
* with a parameter for the function to use on the related model

Reply:
Not to be done in this PR.
to put on backlog

Reply:
To be done in the future